contractcourt/commit_sweep_resolver.go

```diff
@@ -86,127 +78,101 @@ func (c *commitSweepResolver) Resolve() (ContractResolver, error) {
-				c, c.chanPoint, err)
-			return nil, err
-		}
+		c.resolved = true
+		return nil, c.Checkpoint(c)
+	}
 
-	// Otherwise, this is our commitment transaction, So we'll obtain the
-	// sweep transaction once the commitment output has been spent.
-	case c.sweepTx == nil && isLocalCommitTx:
-		// Otherwise, if we're dealing with our local commitment
-		// transaction, then the output we need to sweep has been sent
-		// to the nursery for incubation. In this case, we'll wait
-		// until the commitment output has been spent.
-		spendNtfn, err := c.Notifier.RegisterSpendNtfn(
-			&c.commitResolution.SelfOutPoint,
-			c.commitResolution.SelfOutputSignDesc.Output.PkScript,
-			c.broadcastHeight,
-		)
-		if err != nil {
-			return nil, err
-		}
+	// Otherwise we are dealing with a local commitment transaction and the
+	// output we need to sweep has been sent to the nursery for incubation.
+	// In this case, we'll wait until the commitment output has been spent.
+	spendNtfn, err := c.Notifier.RegisterSpendNtfn(
+		&c.commitResolution.SelfOutPoint,
+		c.commitResolution.SelfOutputSignDesc.Output.PkScript,
+		c.broadcastHeight,
+	)
+	if err != nil {
+		return nil, err
+	}
 
-		log.Infof("%T(%v): waiting for commit output to be swept", c,
-			c.chanPoint)
+	log.Infof("%T(%v): waiting for commit output to be swept", c,
+		c.chanPoint)
 
-		select {
-		case commitSpend, ok := <-spendNtfn.Spend:
-			if !ok {
-				return nil, fmt.Errorf("quitting")
-			}
+	var sweepTx *wire.MsgTx
+	select {
+	case commitSpend, ok := <-spendNtfn.Spend:
+		if !ok {
+			return nil, fmt.Errorf("quitting")
+		}
 
-			// Once we detect the commitment output has been spent,
-			// we'll extract the spending transaction itself, as we
-			// now consider this to be our sweep transaction.
-			c.sweepTx = commitSpend.SpendingTx
+		// Once we detect the commitment output has been spent,
+		// we'll extract the spending transaction itself, as we
+		// now consider this to be our sweep transaction.
+		sweepTx = commitSpend.SpendingTx
 
-			log.Infof("%T(%v): commit output swept by txid=%v",
-				c, c.chanPoint, c.sweepTx.TxHash())
+		log.Infof("%T(%v): commit output swept by txid=%v",
+			c, c.chanPoint, sweepTx.TxHash())
 
-			if err := c.Checkpoint(c); err != nil {
-				log.Errorf("unable to Checkpoint: %v", err)
-				return nil, err
-			}
-		case <-c.Quit:
-			return nil, fmt.Errorf("quitting")
+		if err := c.Checkpoint(c); err != nil {
+			log.Errorf("unable to Checkpoint: %v", err)
+			return nil, err
 		}
+	case <-c.Quit:
+		return nil, fmt.Errorf("quitting")
 	}
 
 	log.Infof("%T(%v): waiting for commit sweep txid=%v conf", c, c.chanPoint,
-		c.sweepTx.TxHash())
+		sweepTx.TxHash())
 
 	// Now we'll wait until the sweeping transaction has been fully
 	// confirmed.  Once it's confirmed, we can mark this contract resolved.
-	sweepTXID := c.sweepTx.TxHash()
-	sweepingScript := c.sweepTx.TxOut[0].PkScript
+	sweepTXID := sweepTx.TxHash()
+	sweepingScript := sweepTx.TxOut[0].PkScript
```

Comment:
iirc, the spend notification will only trigger after it has one conf, so i'm not sure this second stage does anything? obviously this existed prior and independent of this PR, but perhaps we should consider bumping the number of confirmations

Reply:
I indeed think this second stage does nothing, but not sure if we want to change anything to it in this pr. The greater plan is to remove nursery completely and wait out the csv inside the resolver.

Reply:
Yeah this was left over from when we still did mempool confirmations. IMO there's no harm in leaving it in as we should move away from single conf dispatch in many areas of the codebase.